unifyfs@0.9.2 %oneapi build fails: unifyfs.c:1502:7: error: unused function 'next_page_align' [-Werror,-Wunused-function]

### Steps to reproduce the issue

`unifyfs@0.9.2 %oneapi@2022.1.0` build is failing using:
* `spack/develop` (ec879240396b457f8bf05361a7d55280e0fa19fb from `Fri Jul 29 19:12:02 2022 -0700`)
* Ubuntu 20.04, OneAPI 2022.1.0 (installed via offline installer)
* Docker container image `ecpe4s/ubuntu20.04-runner-x86_64-oneapi:2022-07-01` (OneAPI pre-installed)

This issue is coming up in our E4S OneAPI CI stack:
* https://github.com/spack/spack/blob/develop/share/spack/gitlab/cloud_pipelines/stacks/e4s-oneapi/spack.yaml#L212
* https://github.com/spack/spack/blob/develop/share/spack/gitlab/cloud_pipelines/stacks/e4s-oneapi/spack.yaml#L212

Concrete spec: [unifyfs.spec.json.txt](https://github.com/spack/spack/files/9227019/unifyfs.spec.json.txt)

Concretization:
```
 -   unifyfs@0.9.2%oneapi@2022.1.0+auto-mount~fortran~pmi~pmix+spath patches=8a9c20c arch=linux-ubuntu20.04-x86_64
[+]      ^autoconf@2.69%oneapi@2022.1.0 patches=35c4492,7793209,a49dd5b arch=linux-ubuntu20.04-x86_64
[+]          ^m4@1.4.19%oneapi@2022.1.0+sigsegv patches=9dc5fbd,bfdffa7 arch=linux-ubuntu20.04-x86_64
[+]              ^diffutils@3.8%oneapi@2022.1.0 arch=linux-ubuntu20.04-x86_64
[+]                  ^libiconv@1.16%oneapi@2022.1.0 libs=shared,static arch=linux-ubuntu20.04-x86_64
[+]              ^libsigsegv@2.13%oneapi@2022.1.0 arch=linux-ubuntu20.04-x86_64
[+]          ^perl@5.34.1%oneapi@2022.1.0+cpanm+shared+threads arch=linux-ubuntu20.04-x86_64
[+]              ^berkeley-db@18.1.40%oneapi@2022.1.0+cxx~docs+stl patches=b231fcc arch=linux-ubuntu20.04-x86_64
[+]              ^bzip2@1.0.8%oneapi@2022.1.0~debug~pic+shared arch=linux-ubuntu20.04-x86_64
[+]              ^gdbm@1.19%oneapi@2022.1.0 arch=linux-ubuntu20.04-x86_64
[+]                  ^readline@8.1.2%oneapi@2022.1.0 arch=linux-ubuntu20.04-x86_64
[+]                      ^ncurses@6.2%oneapi@2022.1.0~symlinks+termlib abi=none arch=linux-ubuntu20.04-x86_64
[+]                          ^pkgconf@1.8.0%oneapi@2022.1.0 arch=linux-ubuntu20.04-x86_64
[+]              ^zlib@1.2.12%oneapi@2022.1.0+optimize+pic+shared patches=0d38234 arch=linux-ubuntu20.04-x86_64
[+]      ^automake@1.16.5%oneapi@2022.1.0 arch=linux-ubuntu20.04-x86_64
[+]      ^gotcha@1.0.3%oneapi@2022.1.0~ipo~test build_type=RelWithDebInfo arch=linux-ubuntu20.04-x86_64
[+]          ^cmake@3.23.2%oneapi@2022.1.0~doc+ncurses+ownlibs~qt build_type=Release arch=linux-ubuntu20.04-x86_64
[+]              ^openssl@1.1.1q%oneapi@2022.1.0~docs~shared certs=mozilla patches=3fdcf2d arch=linux-ubuntu20.04-x86_64
[+]                  ^ca-certificates-mozilla@2022-03-29%oneapi@2022.1.0 arch=linux-ubuntu20.04-x86_64
[+]      ^libfabric@1.14.1%oneapi@2022.1.0~debug~kdreg fabrics=rxm,sockets,tcp,udp arch=linux-ubuntu20.04-x86_64
[+]      ^libtool@2.4.7%oneapi@2022.1.0 arch=linux-ubuntu20.04-x86_64
[+]      ^mochi-margo@0.9.10%oneapi@2022.1.0 arch=linux-ubuntu20.04-x86_64
[+]          ^argobots@1.1%oneapi@2022.1.0~affinity~debug+perf~stackunwind~tool~valgrind stackguard=none arch=linux-ubuntu20.04-x86_64
[+]          ^json-c@0.16%oneapi@2022.1.0~ipo build_type=RelWithDebInfo arch=linux-ubuntu20.04-x86_64
[+]          ^mercury@2.1.0%oneapi@2022.1.0~bmi+boostsys+checksum~debug~ipo+mpi+ofi+shared+sm~ucx~udreg build_type=RelWithDebInfo arch=linux-ubuntu20.04-x86_64
[+]              ^boost@1.79.0%oneapi@2022.1.0+atomic+chrono~clanglibcpp~container~context~contract~coroutine+date_time~debug+exception~fiber+filesystem+graph~graph_parallel~icu+iostreams~json+locale+log+math+mpi+multithreaded~nowide~numpy~pic+program_options~python+random+regex+serialization+shared+signals~singlethreaded~stacktrace+system~taggedlayout+test+thread+timer~type_erasure~versionedlayout+wave cxxstd=98 patches=a440f96,b8569d7 visibility=hidden arch=linux-ubuntu20.04-x86_64
[+]                  ^mpich@4.0.2%oneapi@2022.1.0~argobots~cuda+fortran+hwloc+hydra+libxml2+pci~rocm+romio~slurm~two_level_namespace~vci~verbs~wrapperrpath datatype-engine=auto device=ch4 netmod=ofi patches=b4c706e pmi=pmi arch=linux-ubuntu20.04-x86_64
[+]                      ^findutils@4.9.0%oneapi@2022.1.0 patches=440b954 arch=linux-ubuntu20.04-x86_64
[+]                      ^hwloc@2.7.1%oneapi@2022.1.0~cairo~cuda~gl~libudev+libxml2~netloc~nvml~oneapi-level-zero~opencl+pci~rocm+shared arch=linux-ubuntu20.04-x86_64
[+]                          ^libpciaccess@0.16%oneapi@2022.1.0 arch=linux-ubuntu20.04-x86_64
[+]                              ^util-macros@1.19.3%oneapi@2022.1.0 arch=linux-ubuntu20.04-x86_64
[+]                          ^libxml2@2.9.13%oneapi@2022.1.0~python arch=linux-ubuntu20.04-x86_64
[+]                              ^xz@5.2.5%oneapi@2022.1.0+pic libs=shared,static arch=linux-ubuntu20.04-x86_64
[+]                      ^yaksa@0.2%oneapi@2022.1.0~cuda~rocm arch=linux-ubuntu20.04-x86_64
[+]                          ^python@3.8.13%oneapi@2022.1.0+bz2+ctypes+dbm~debug+ensurepip+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tix~tkinter~ucs4+uuid+zlib patches=0d98e93,4c24573,f2fd060 arch=linux-ubuntu20.04-x86_64
[+]                              ^expat@2.4.8%oneapi@2022.1.0+libbsd arch=linux-ubuntu20.04-x86_64
[+]                                  ^libbsd@0.11.5%oneapi@2022.1.0 arch=linux-ubuntu20.04-x86_64
[+]                                      ^libmd@1.0.4%oneapi@2022.1.0 arch=linux-ubuntu20.04-x86_64
[+]                              ^gettext@0.21%oneapi@2022.1.0+bzip2+curses+git~libunistring+libxml2+tar+xz arch=linux-ubuntu20.04-x86_64
[+]                                  ^tar@1.34%oneapi@2022.1.0 zip=pigz arch=linux-ubuntu20.04-x86_64
[+]                                      ^pigz@2.7%oneapi@2022.1.0 arch=linux-ubuntu20.04-x86_64
[+]                                      ^zstd@1.5.2%oneapi@2022.1.0+programs compression=none libs=shared,static arch=linux-ubuntu20.04-x86_64
[+]                              ^libffi@3.4.2%oneapi@2022.1.0 arch=linux-ubuntu20.04-x86_64
[+]                              ^sqlite@3.38.5%oneapi@2022.1.0+column_metadata+dynamic_extensions+fts~functions+rtree arch=linux-ubuntu20.04-x86_64
[+]                              ^util-linux-uuid@2.37.4%oneapi@2022.1.0 arch=linux-ubuntu20.04-x86_64
[+]      ^spath@0.2.0%oneapi@2022.1.0~ipo~mpi+shared build_type=RelWithDebInfo arch=linux-ubuntu20.04-x86_64
```

Install error:
```
==> Installing unifyfs-0.9.2-2rjepszwaxvvp7swuewlvzzn2d6jffib
==> No binary for unifyfs-0.9.2-2rjepszwaxvvp7swuewlvzzn2d6jffib found: installing from source
==> Using cached archive: /spack/var/spack/cache/_source-cache/archive/70/7046625dc0677535f5d960187cb2e2d58a6f8cfb4dc6a3604f825257eb0891aa.tar.gz
==> Applied patch /spack/var/spack/repos/builtin/packages/unifyfs/include-sys-sysmacros.h.patch
==> unifyfs: Executing phase: 'autoreconf'
==> unifyfs: Executing phase: 'configure'
==> unifyfs: Executing phase: 'build'
...
  >> 436    /tmp/root/spack-stage/spack-stage-unifyfs-0.9.2-2rjepszwaxvvp7swuewlvzzn2d6jffib/spack-src/client/src/unifyfs.c:1502:7: error: unused function 'next_
            page_align' [-Werror,-Wunused-function]
     437    char* next_page_align(char* ptr)
     438          ^
     439    1 error generated.
  >> 440    make[3]: *** [Makefile:938: libunifyfs_la-unifyfs.lo] Error 1
     441    make[3]: Leaving directory '/tmp/root/spack-stage/spack-stage-unifyfs-0.9.2-2rjepszwaxvvp7swuewlvzzn2d6jffib/spack-src/spack-build/client/src'
  >> 442    make[2]: *** [Makefile:469: all-recursive] Error 1
     443    make[2]: Leaving directory '/tmp/root/spack-stage/spack-stage-unifyfs-0.9.2-2rjepszwaxvvp7swuewlvzzn2d6jffib/spack-src/spack-build/client'
  >> 444    make[1]: *** [Makefile:497: all-recursive] Error 1
     445    make[1]: Leaving directory '/tmp/root/spack-stage/spack-stage-unifyfs-0.9.2-2rjepszwaxvvp7swuewlvzzn2d6jffib/spack-src/spack-build'
  >> 446    make: *** [Makefile:408: all] Error 2
```



### Error message

<details><summary>Error message</summary><pre>
... see above
</pre></details>


### Information on your system

* **Spack:** 0.19.0.dev0 (ec879240396b457f8bf05361a7d55280e0fa19fb)
* **Python:** 3.8.10
* **Platform:** linux-ubuntu20.04-cascadelake
* **Concretizer:** clingo

### Additional information

[spack-build-out.txt](https://github.com/spack/spack/files/9227023/spack-build-out.txt)

@CamStan @wspear 

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/share/spack/gitlab/cloud_pipelines/stacks/e4s-oneapi/spack.yaml
+++ b/share/spack/gitlab/cloud_pipelines/stacks/e4s-oneapi/spack.yaml
@@ -177,6 +177,7 @@ spack:
   - turbine
   - umap
   - umpire
+  - unifyfs
   - upcxx
   - veloc
   - vtk-m ~openmp # can't build +openmp w/ %oneapi: https://github.com/spack/spack/issues/31830
@@ -210,7 +211,6 @@ spack:
   #- py-warpx@22.05 ^warpx dims=2                     # adios2
   #- py-warpx@22.05 ^warpx dims=3                     # adios2
   #- py-warpx@22.05 ^warpx dims=rz                    # adios2
-  #- unifyfs@0.9.2                                    # unifyfs
   #- variorum@0.4.1                                   # variorum
 
   # CPU BUILD FAILURES - NOTES
@@ -229,7 +229,6 @@ spack:
   # ruby: limits.c:415:34: error: invalid suffix 'D' on floating constant
   # rust: /usr/bin/ld: /opt/intel/oneapi/compiler/2022.1.0/linux/bin-llvm/../compiler/lib/intel64_lin/libimf.a(libm_feature_flag.o): in function `__libm_feature_flag_init': libm_feature_flag.c:(.text+0x25): undefined reference to `__intel_cpu_feature_indicator_x'
   # scr: scr_globals.h:81:10: fatal error: 'spath_mpi.h' file not found: #include "spath_mpi.h"
-  # unifyfs: client/src/unifyfs.c:1502:7: error: unused function 'next_page_align' [-Werror,-Wunused-function]
   # variorum: ld: Intel/CMakeFiles/variorum_intel.dir/msr_core.c.o:(.bss+0x0): multiple definition of `g_platform'; CMakeFiles/variorum.dir/config_architecture.c.o:(.bss+0x0): first defined here
   # vtk-m: clang++: error: clang frontend command failed with exit code 139 (use -v to see invocation)
 

--- a/var/spack/repos/builtin/packages/unifyfs/package.py
+++ b/var/spack/repos/builtin/packages/unifyfs/package.py
@@ -84,6 +84,8 @@ class Unifyfs(AutotoolsPackage):
         # See https://gcc.gnu.org/bugzilla/show_bug.cgi?id=98266
         if "%gcc@11" in self.spec:
             env.append_flags("CFLAGS", "-Wno-array-bounds")
+        if self.spec.satisfies("%oneapi"):
+            env.append_flags("CFLAGS", "-Wno-unused-function")
 
     def configure_args(self):
         spec = self.spec

